airflow/providers/databricks/operators/databricks.py

```diff
@@ -521,7 +521,7 @@ def execute(self, context: Context):
         ):
             # If pipeline_id is not provided, we need to fetch it from the pipeline_name
             pipeline_name = self.json["pipeline_task"]["pipeline_name"]
-            self.json["pipeline_task"]["pipeline_id"] = self._hook.get_pipeline_id(pipeline_name)
```

Comment:
I wonder why Mypy didn't block #32903 which added this call to the non-existing method. @potiuk do you have an idea?

Reply:
It's a mistery . MyPy uses a number of heuristics to find types of particular objects and sometime it gives up or sometimes drop types if they are too complex to analyse. Generally MyPy's behaveriour is often non-deterministic/difficult to reason about.